migrate org-specific recipes from meta-orgconf

amstewart opened the pull request:
The meta-orgconf layer is being deprecated. Unfortunately, there are
still some recipes which rely on org-specific information to build -
particularly the recovery media.

Move the required classes and recipes from meta-orgconf into meta-nilrt,
so that we can deprecate the former.

This commit does not modify the contents of the moved files.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

----

I was under the mistaken impression that these files were only used by NXG; they are necessary for os-common to build the recovery media.

## Testing
* os-common/8.8 now builds on my dev machine (with necessary fixups to the p4 components).

@ni/rtos

amstewart committed:
migrate org-specific recipes from meta-orgconf

The meta-orgconf layer is being deprecated. Unfortunately, there are
still some recipes which rely on org-specific information to build -
particularly the recovery media.

Move the required classes and recipes from meta-orgconf into meta-nilrt,
so that we can deprecate the former.

This commit does not modify the contents of the moved files.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

shruthi-ravi commented on recipes-org/zynq-safemode-cfgs/zynq-safemode-cfgs.bb:
```diff
@@ -0,0 +1,41 @@
+SUMMARY = "Packages containing older nilrd ARM .cfg files + a script to install them"
```
How's this recipe used?

amstewart replied:
I'm not actually sure. [Here's the history](https://github.com/ni/meta-orgconf/commits/nilrt/20.7/sumo/recipes-core/zynq-safemode-cfgs). I'm not confident enough with why it was added to assert that it should be removed.

shruthi-ravi replied:
Hmmm. Does os-common build without it? I didn't see any codesearch references for this recipe.

> zynq-safemode-cfgs: add new recipe
This recipe creates safemode packages for restoring all nilrt supported
devices on the older OS version. The contents of each package is a *.cfg
file containing the necessary files for installation (itb, bitfiles,
preinst, postinst) and a standard script to install a cfg file.
These ipks will be downloaded on target from feeds during backwards
migration on arm devices based on their device code.

Is the package even added to our feeds? If the answers are no, and no, let's remove it?

amstewart committed:
recipes-ni: deprecate zynq-safemode-cfgs

I think the zynq-safemode-cfg recipes were necessary at some point in
the past, as a way to enable feed-based migration of ARM targets. That
feature has been deprecated for years, and I can find no reference to
any component which builds or uses these IPKs.

Deprecate the recipes and remove them from the layer, for cleanliness.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
## Update 2
* After searching around more thoroughly for anything which uses either the `zynq-safemode-cfg` IPKS - or their contents - I cannot find any users. Added a commit to this PR which deprecates the cfg recipes.
* Rebased on `nilrt/master/sumo`.

shruthi-ravi commented on recipes-org/images/safemode-image.bb:
```diff
@@ -0,0 +1,59 @@
+        mkdir -p ${BS_EXPORT_DATA}
+
+        if [ -z "$SAFEMODE_PAYLOAD_PATH" ]; then
+                EXPORTS_TO_FETCH=$(bs_get_latest_export "nilinux/os-common/export/8.0")"/standard_x64_safemode.tar.gz"
```
Does the version in this path have to be updated every release?
